### ISSUE TITLE
Microsoft Graph race conditions

### ISSUE THREAD

**Issue by bmc-msft** [microsoft/onefuzz#938]
Prior to moving from AAD graph to Microsoft Graph, delays from creating objects to the objects being accessible by other parts of Azure were caught and worked around via retry.  With the recent move in OneFuzz to Microsoft Graph, it appears we're missing events that need to be retried.

```
INFO:deploy:checking if RBAC already exists
INFO:deploy:creating Application registration
INFO:deploy:creating service principal
Traceback (most recent call last):
  File "deploy.py", line 1087, in <module>
    main()
  File "deploy.py", line 1081, in main
    state[1](client)
  File "deploy.py", line 408, in setup_rbac
    (password_id, password) = self.create_password(app.object_id)
  File "deploy.py", line 251, in create_password
    return add_application_password(object_id, self.get_subscription_id())
  File "/tmp/tmp7khuh4k8/registration.py", line 272, in add_application_password
    raise error
  File "/tmp/tmp7khuh4k8/registration.py", line 256, in add_application_password
    password = add_application_password_impl(app_object_id, subscription_id)
  File "/tmp/tmp7khuh4k8/registration.py", line 315, in add_application_password_impl
    password: Dict = query_microsoft_graph(
  File "/tmp/tmp7khuh4k8/registration.py", line 77, in query_microsoft_graph
    raise GraphQueryError(
registration.GraphQueryError: request did not succeed: HTTP 404 - {"error":{"code":"Request_ResourceNotFound","message":"Resource 'eee419bf-1f7b-4aec-bab5-8682d95bff10' does not exist or one of its queried reference-property objects are not present.","innerError":{"date":"2021-05-28T15:41:13","request-id":"b8c7022b-c62c-4f4b-abbd-6de890463c21","client-request-id":"b8c7022b-c62c-4f4b-abbd-6de890463c21"}}}
```

**Comment by bmc-msft** [comment]
CC: @chkeita

**Comment by bmc-msft** [comment]
This looks to probably be fixed in #947 though #952 will address logging issues here.

**Comment by bmc-msft** [comment]
Addressing the issues discussed in the previous comment highlighted that we need to add retry to the `authorize_application` calls to query_microsoft_graph.


```
INFO:deploy:checking if RBAC already exists
INFO:deploy:creating Application registration                                                                                             INFO:deploy:creating service principal                                                                                                    INFO:deploy:app unavailable in AAD, unable to create password yet
INFO:deploy:retrying app password creation
INFO:deploy:app unavailable in AAD, unable to create password yet
INFO:deploy:retrying app password creation                                                                                                INFO:deploy:app unavailable in AAD, unable to create password yet                                                                         INFO:deploy:retrying app password creation
INFO:deploy:app unavailable in AAD, unable to create password yet
INFO:deploy:retrying app password creation
INFO:deploy:app unavailable in AAD, unable to create password yet                                                                         INFO:deploy:retrying app password creation                                                                                                INFO:deploy:app password created
Traceback (most recent call last):
  File "deploy.py", line 1087, in <module>
    main()                                                                                                                                  File "deploy.py", line 1081, in main                                                                                                        state[1](client)                                                                                                                        File "deploy.py", line 436, in setup_rbac                                                                                                   authorize_application(uuid.UUID(onefuzz_cli_app.app_id), app.app_id)                                                                    File "/tmp/tmp975fzhjc/registration.py", line 370, in authorize_application                                                                 query_microsoft_graph(
  File "/tmp/tmp975fzhjc/registration.py", line 77, in query_microsoft_graph
    raise GraphQueryError(
registration.GraphQueryError: request did not succeed: HTTP 404 - {"error":{"code":"Request_ResourceNotFound","message":"Resource '69b3235f-0ceb-4ccf-b1c6-8839b0da2b71' does not exist or one of its queried reference-property objects are not present.","innerError":{"date":"2021-06-02T19:46:45","request-id":"a40c7158-37f4-4141-93ba-6dc369d0149f","client-request-id":"a40c7158-37f4-4141-93ba-6dc369d0149f"}}}
```
